### PR TITLE
Feat: wallet connect reconnect

### PR DESCRIPTION
**Pull request by Asmadek**
image

### CODE REVIEW

**Comment by Asmadek** [comment]
Task linked: [CU-8692u93cr Refresh Nova Wallet & WalletConnect wallet from details](https://app.clickup.com/t/8692u93cr)

**Review by egor0798 — APPROVED** [review]

File: src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx [hunk 1]
```diff
@@ -28,17 +45,27 @@ export const WalletConnectDetails = ({ isOpen, wallet, accounts, onClose }: Prop
...
   const connected = useUnit(walletProviderModel.$connected);
+  const reset = useUnit(walletProviderModel.events.reset);
+  const connect = useUnit(walletConnectModel.events.connect);
+  const reconnectStarted = useUnit(walletProviderModel.events.reconnectStarted);
+  const reconnectAborted = useUnit(walletProviderModel.events.reconnectAborted);
+  const reconnectStep = useUnit(walletProviderModel.$reconnectStep);
 
-  const closeWowModal = () => {
+  useEffect(() => {
+    if (isModalOpen) {
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx:55]: Looks to me that `reset` should be called on modal close
When model initiates we have a clean state, so there's actually nothing to reset
When we leave we reset so next time when open it's already clean

File: src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx [hunk 2]
```diff
@@ -55,15 +82,53 @@ export const WalletConnectDetails = ({ isOpen, wallet, accounts, onClose }: Prop
...
+  const reconnect = () => {
+    connect({
+      chains: getWalletConnectChains(chainsService.getChainsData()),
+      pairing: { topic: accounts[0].signingExtras?.pairingTopic },
+    });
+
+    reconnectStarted();
+  };
+
+  const isAccountsStep = reconnectStep === ReconnectStep.NOT_STARTED && connected;
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx:94]: ⚠️ Logic in UI ⚠️
These flags should stay in model which knows how to produce them

File: src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx [hunk 3]
```diff
@@ -79,9 +144,39 @@ export const WalletConnectDetails = ({ isOpen, wallet, accounts, onClose }: Prop
...
+                  <source src={wallet_connect_reconnect_webm} type="video/webm" />
+                  <source src={wallet_connect_reconnect} type="video/mp4" />
+                </video>
+              </div>
+            )}
+          </>
         </div>
+
+        <StatusModal
+          isOpen={reconnectStep === ReconnectStep.REJECTED}
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx:172]: This flag goes to model as well

File: src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx [hunk 4]
```diff
@@ -55,15 +82,53 @@ export const WalletConnectDetails = ({ isOpen, wallet, accounts, onClose }: Prop
     return accountsList;
   }, []);
 
+  const reconnect = () => {
+    connect({
+      chains: getWalletConnectChains(chainsService.getChainsData()),
+      pairing: { topic: accounts[0].signingExtras?.pairingTopic },
+    });
+
+    reconnectStarted();
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx:91]: Why not call `reconnectStarted` or `reconnected` pass all `connect` arguments there?
In model do reconnect stuff and then connect? 
Right not it's a logic in UI: we init connect (that's fine) but then UI somehow decides to call `reconnectStarted`

File: src/renderer/entities/walletConnect/model/wallet-connect-model.ts [hunk 5]
```diff
@@ -33,17 +33,18 @@ type ConnectResult = {
 };
 
 type SessionTopicUpdateProps = {
```
Comment (tuul-wq) [inline src/renderer/entities/walletConnect/model/wallet-connect-model.ts:35]: I'd rather name it `SessionTopicParams`, `Props` is a react-world naming
effector is framework agnostic 😉

**Review by tuul-wq — CHANGES_REQUESTED** [review]

File: src/renderer/features/operation/sign/ui/WalletConnect/WalletConnect.tsx [hunk 6]
```diff
@@ -67,7 +67,7 @@ export const WalletConnect = ({ api, validateBalance, onGoBack, accounts, transa
       setIsNeedUpdate(false);
 
       if (session?.topic) {
-        walletConnectModel.events.sessionTopicUpdated(session?.topic);
+        walletConnectModel.events.currentSessionTopicUpdated(session?.topic);
```
[the remark this answers is not in the document]
Comment (Asmadek) [reply src/renderer/features/operation/sign/ui/WalletConnect/WalletConnect.tsx:70]: Right, we have `sessionTopicUpdated`, where we should provide topic and accounts, and `currentSessionTopicUpdated` where we provide only topic

File: src/renderer/widgets/WalletDetails/common/const.ts
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/common/const.ts:1]: Let's move this file to `lib/constans.ts` according to FSD

**Review by tuul-wq — CHANGES_REQUESTED** [review]

File: src/renderer/widgets/WalletDetails/model/wallet-connect-details-model.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+sample({
+  clock: walletConnectModel.events.connected,
+  source: {
+    accounts: walletProviderModel.$accounts,
+    session: walletConnectModel.$session,
+  },
+
+  filter: ({ accounts, session }) => accounts.length > 0 && Boolean(session?.topic),
+  fn: ({ accounts, session }) => ({
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/model/wallet-connect-details-model.ts:45]: I guess this should work
```suggestion
fn: ({ accounts, session }) => ({ accounts, topic: session!.topic }),
```

**Review by tuul-wq — CHANGES_REQUESTED** [review]
A few comments to increase readability and maintainability 😉

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**